clickhouse-http-client/src/main/java/com/clickhouse/client/http/ClickHouseHttpConnection.java

```diff
@@ -118,10 +118,25 @@ static String buildQueryParams(ClickHouseRequest<?> request, Map<String, Seriali
             appendQueryParameter(builder, settingKey, String.valueOf(config.getMaxExecutionTime()));
         }
         settingKey = ClickHouseClientOption.MAX_RESULT_ROWS.getKey();
-        if (config.getMaxResultRows() > 0L && !settings.containsKey(settingKey)) {
+        boolean hasRequestSetting = settings.containsKey(settingKey);
+        if (config.getMaxResultRows() > 0L && !hasRequestSetting) {
+            // set on client level
             appendQueryParameter(builder, settingKey, String.valueOf(config.getMaxResultRows()));
-            appendQueryParameter(builder, "result_overflow_mode", "break");
+        } else if (hasRequestSetting) {
+            // set on request level
+            Object value = settings.get(settingKey);
+            if (value instanceof Number && ((Number) value).longValue() > 0L) {
```

Comment:
Why are we not checking the type on insert into the settings so we will not have instanceof on each time we build settings

Reply:
This settings are coming from different places and it may be Number or String. I've made this check only for safety reasons.

Reply:
Just saying we can make this check/conversion when the settings are updated, so when we run this method, we will deal with it, but correct me here it seems like it will deprecated and remove this part of code

Reply:
Yes, this will gone soon.